modules/renter/siafile/siafile.go

```diff
@@ -0,0 +1,181 @@
+package siafile
+
+import (
+	"bytes"
+	"encoding/base32"
+	"encoding/binary"
+	"fmt"
+	"sync"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+	"github.com/NebulousLabs/fastrand"
+
+	"github.com/NebulousLabs/Sia/crypto"
+)
+
+type (
+	// SiaFile is the disk format for files uploaded to the Sia network.  It
+	// contains all the necessary information to recover a file from its hosts and
+	// allows for easy constant-time updates of the file without having to read or
+	// write the whole file.
+	SiaFile struct {
+		// metadata is the mostly static metadata of a SiaFile. The reserved
+		// size of the metadata on disk should always be a multiple of 4kib.
+		// The metadata is also the only part of the file that is JSON encoded
+		// and can therefore be easily extended.
+		metadata Metadata
+
+		// pubKeyTable stores the public keys of the hosts this file's pieces are uploaded to.
+		// Since multiple pieces from different chunks might be uploaded to the same host, this
+		// allows us to deduplicate the rather large public keys.
+		pubKeyTable []types.SiaPublicKey
+
+		// chunks are the chunks the file was split into.
+		chunks []Chunk
+
+		// utility fields. These are not persisted.
+		deleted     bool
+		erasureCode modules.ErasureCoder
```

Comment:
That's a good point. I'll change it so that it always uses `siafile.chunks[0].erasureCode` but we need to think about how we would like to report the redundancy on a file that has different settings for each chunk.